s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/config/S3SourceConfig.java

```diff
@@ -179,6 +173,11 @@ private static void addOtherConfig(final S3SourceConfigDef configDef) {
                 "Value converter", GROUP_OTHER, awsOtherGroupCounter++, // NOPMD
                 // UnusedAssignment
                 ConfigDef.Width.NONE, VALUE_CONVERTER);
+        configDef.define(MAX_MESSAGE_BYTES_SIZE, ConfigDef.Type.INT, 1_048_588, ConfigDef.Importance.MEDIUM,
```

Comment:
This is a very confusing name, because it's not doing the same thing as Kafka's configuration.

It's only active for the ByteArrayTransformer, and is a "chunk size" for breaking up large files into multiple messages.

Without some metadata about where a particular chunk of a file is from, or an indication of when a file is truncated, I think this configuration is a huge footgun. Users will get one message of the first 1MB of their file, and may not know that the file was actually larger.

This needs more thought.

Reply:
Further implementation wil be handled in ticket https://aiven.atlassian.net/browse/KCON-3